codex/slots/proofs/backends/converters.nim

```diff
@@ -52,3 +54,32 @@ func toG2*(g: CircomG2): G2Point =
 
 func toGroth16Proof*(proof: CircomProof): Groth16Proof =
   Groth16Proof(a: proof.a.toG1, b: proof.b.toG2, c: proof.c.toG1)
+
+proc newProof*(): ptr Proof =
```

Comment:
I think it would be better if we declare this as types and use the standard `new` conventino to construct them.

```nim
type
  ProofPtr = ptr Proof

proc new(_: ProofPtr): ProofPtr = 
 ...
```

Same for the verify types.